src/main/java/org/datadog/jmxfetch/App.java

```diff
@@ -244,14 +245,25 @@ void start() {
                 adPipe = newAutoDiscoveryPipe();
             }
             try {
-                int len;
-                if(adPipe != null && (len = adPipe.available()) > 0) {
-                    byte[] buffer = new byte[len];
-                    adPipe.read(buffer);
+                if(adPipe != null) {
+                    int offset = 0;
+                    byte[] buffer = new byte[0];
+                    while (adPipe.available() > 0) {
```

Comment:
I'd have to look at the internals, I'm not sure. I haven't encountered issues while testing, but this is a valid point.